File: catkin_ws/src/clear_map_service/src/clear_map_service_client.cpp
#include "ros/ros.h"
#include "clear_map_service/GetMap.h"
#include "nav_msgs/GetMap.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "clear_map_service_client");

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<clear_map_service::GetMap>("clear_map_service");
  clear_map_service::GetMap srv;
  nav_msgs::OccupancyGrid newMap;

  ros::Publisher commandPub = n.advertise<nav_msgs::OccupancyGrid>("/new_map", 10);

  ros::Rate rate(1);
  ROS_INFO("Publishing new map to topic /new_map");
  while(ros::ok())
  {
    if (client.call(srv))
    {
      newMap = srv.response.map;
      ROS_INFO("I have got a map from server. Ready to publish");
      ROS_INFO("Publishing");
      commandPub.publish(newMap);
    }
    else
    {
      ROS_ERROR("Failed to call service");
      return 1;
    }

    ros::spinOnce();
    rate.sleep();
  }

  return 0;
}

File: catkin_ws/src/clear_map_service/scripts/clear_map_service_client.py
#!/usr/bin/env python

import sys
import rospy
from nav_msgs.msg import OccupancyGrid
from clear_map_service.srv import GetMap

def clear_map_service_client():
	rospy.init_node('clear_map_service_client', anonymous=False)
	pub = rospy.Publisher('/new_map', OccupancyGrid, queue_size=10)
	print('Publishing new map to topic /new_map')
	rate = rospy.Rate(1)
	while not rospy.is_shutdown():
		rospy.wait_for_service('clear_map_service')
		try:
			getNewMap = rospy.ServiceProxy('clear_map_service', GetMap)
			response = getNewMap().map
			print('I have got a map from server. Ready to publish')
		except rospy.ServiceException as e:
			print("Service call failed: %s"%e)
		print('Publishing')
		pub.publish(response)
		rate.sleep()

if __name__ == "__main__":       
	clear_map_service_client()
